packages/Python/lldbsuite/test/lang/swift/expression/exclusivity_suppression/TestExclusivitySuppression.py

```diff
@@ -0,0 +1,145 @@
+        frame = thread.frames[0]
+        self.assertTrue(frame, "Frame 0 is valid.")
+
+        self.check_expression(frame, "w.s.i", "8", use_summary=False)
+
+    # Test that we properly handle nested expression evaluations by:
+    # (1) Breaking at breakpoint 1
+    # (2) Running 'expr get()' (which will hit breakpoint 2)
+    # (3) Evaluating w.s.i at breakpoint 2 (this is a nested evaluation)
+    # (4) Contininging the evaluation of 'expr get()' to return to bp 1
+    # (5) Evaluating w.s.i again to check that finishing the nested expression
+    #     did not prematurely re-enable exclusivity checks.
+    @decorators.swiftTest
+    def test_exclusivity_suppression_for_concurrent_expressions(self):
+        """Test that exclusively owned values can still be accessed"""
+        self.buildAll()
+
+        target = self.create_target()
+
+        bp1 = target.BreakpointCreateBySourceRegex('Breakpoint 1',
+                                                   self.main_source_spec)
+        self.assertTrue(bp1.GetNumLocations() > 0, VALID_BREAKPOINT)
+
+        bp2 = target.BreakpointCreateBySourceRegex('Breakpoint 2',
+                                                   self.main_source_spec)
+        self.assertTrue(bp2.GetNumLocations() > 0, VALID_BREAKPOINT)
+
+        # Launch the process, and do not stop at the entry point.
+        process = target.LaunchSimple(None, None, os.getcwd())
+
+        self.assertTrue(process, PROCESS_IS_VALID)
+
+        # Frame #0 should be at our breakpoint.
+        threads = lldbutil.get_threads_stopped_at_breakpoint(process, bp1)
+
+        self.assertTrue(len(threads) == 1)
+        thread = threads[0]
+        frame = thread.frames[0]
+        self.assertTrue(frame, "Frame 0 is valid.")
+
+        self.check_expression(frame, "w.s.i", "8", use_summary=False)
+
+
+    @decorators.swiftTest
+    def test_exclusivity_suppression_on_expression_bailout(self):
+        """Test that exclusively owned values can still be accessed"""
+        self.buildAll()
+
+
+    def setUp(self):
+        TestBase.setUp(self)
+        self.main_source = "main.swift"
+        self.main_source_spec = lldb.SBFileSpec(self.main_source)
+
+    def buildAll(self):
+        execute_command("make everything")
+        def cleanup():
+            execute_command("make cleanup")
+
+        self.addTearDownHook(cleanup)
+
+    def check_expression(self, frame, expression, expected_result, use_summary=True):
+        value = frame.EvaluateExpression(expression)
+        self.assertTrue(value.IsValid(), expression + "returned a valid value")
+        if self.TraceOn():
+            print value.GetSummary()
+            print value.GetValue()
+        if use_summary:
+            answer = value.GetSummary()
+        else:
+            answer = value.GetValue()
+        report_str = "%s expected: %s got: %s" % (
+            expression, expected_result, answer)
+        self.assertTrue(answer == expected_result, report_str)
+
```

Comment:
Sounds good to me. Is that something that would be better done in a separate commit though?